spec/jobs/verify_fixity_job_spec.rb

```diff
@@ -0,0 +1,217 @@
+          allow(aws_fixity_check).to receive(:fixity_checksum_object_size).and_return ['12345FF', 1234, nil]
+          allow(verify_fixity_job).to receive(:object_checksum_and_size_match?).and_return false
+          expect(aws_fixity_verification_pending).to receive(:failure!)
+          verify_fixity_job.verify_fixity(aws_fixity_verification_pending, aws_fixity_check)
+        end
+      end
+
+      context 'and #object_checksum_and_size_match? returns true' do
+        it 'calls FixityVerfication#failure!' do
+          aws_fixity_check = Atc::Aws::FixityCheck.new(aws_stored_object, 3141)
+          allow(aws_fixity_check).to receive(:fixity_checksum_object_size).and_return ['12345FF', 1234, nil]
+          allow(verify_fixity_job).to receive(:object_checksum_and_size_match?).and_return true
+          expect(aws_fixity_verification_pending).to receive(:success!)
+          verify_fixity_job.verify_fixity(aws_fixity_verification_pending, aws_fixity_check)
+        end
+      end
+    end
+  end
+
+  describe '#object_checksum_and_size_match?' do
+    let(:provider_object_size) { 4321 }
+    let(:provider_object_size_mismatch) { 5321 }
+
+    it 'returns true if checksum and size match' do
+      result = verify_fixity_job.object_checksum_and_size_match?(aws_fixity_verification_pending,
+                                                                 provider_sha256_hex,
+                                                                 provider_object_size)
+      expect(result).to be true
+    end
+
+    it 'returns false if checksums match but sizes do not' do
+      result = verify_fixity_job.object_checksum_and_size_match?(aws_fixity_verification_pending,
+                                                                 provider_sha256_hex,
+                                                                 provider_object_size_mismatch)
+      expect(result).to be false
+    end
+
+    it 'returns false if sizes match but checksums do not' do
+      result = verify_fixity_job.object_checksum_and_size_match?(aws_fixity_verification_pending,
+                                                                 provider_sha256_hex_mismatch,
+                                                                 provider_object_size)
+      expect(result).to be false
+    end
+
+    it 'returns false if both checksums and sizes do not  match' do
+      result = verify_fixity_job.object_checksum_and_size_match?(aws_fixity_verification_pending,
+                                                                 provider_sha256_hex_mismatch,
+                                                                 provider_object_size_mismatch)
+      expect(result).to be false
+    end
+  end
+
+  describe '#checksums_match?' do
+    it 'returns true if checksums match' do
+      result = verify_fixity_job.checksums_match?(atc_sha256_bin, provider_sha256_hex)
+      expect(result).to be true
+    end
+
+    it 'returns false if checksums do notmatch' do
```

Comment:
Really minor thing, missing space: `not match`

Reply:
Typo fixed.